source/contentprovider.d

```diff
@@ -192,34 +200,30 @@ class ContentProvider
 	+/
 	auto getTOC(string language) const
 	{
-		auto chapterTitles = language in chapterTitle_;
-		enforce(chapterTitles !is null, new Exception("%s not known.".format(language)));
-		enforce(ChapterOrdering.length == content_[language].length,
-				new Exception("Chapter ordering doesn't match chapters on disk!"));
-
-		struct Chapter {
-			string title;
-			string chapterId;
-			Tuple!(string, "title", int, "sectionId")[] sections;
-		}
-		
-		auto toc = new Chapter[content_[language].length];
-		
-		foreach (chapterId, sections; content_[language]) {
-			auto idx = ChapterOrdering.countUntil(chapterId);
-			enforce(idx != -1, new Exception("%s chapter not in included in ordering.".format(chapterId)));
-			auto chapter = &toc[idx];
-			chapter.chapterId = chapterId;
-			chapter.title = (*chapterTitles)[chapterId];
-			chapter.sections.length = sections.length;
-			foreach(sectionIdx, ref content; sections) {
-				auto section = &chapter.sections[sectionIdx - 1];
-				section.title = content.title;
-				section.sectionId = sectionIdx;
-			}
-		}
-
-		return toc;
+        struct Chapter {
+            string title;
+            string chapterId;
+            Tuple!(string, "title", string, "sectionId")[] sections;
+        }
+	    Chapter[] toc;
+        auto chapterTitles = language in chapterTitle_;
+        enforce(chapterTitles !is null, new Exception("%s not known.".format(language)));
+
+        // files are already sorted
+        foreach (chapterId, sections; content_[language]) {
```

Comment:
Is this really sorted? _content is a hash map and so doesn't guarantee any ordering.